### PR TITLE
optimizer: fix refiner panic when assert like pattern type to string.

### PR DESCRIPTION
**Pull request by coocood**
The pattern value may not be string.

### CODE REVIEW

File: optimizer/plan/range.go [hunk 1]
```diff
@@ -272,7 +272,11 @@ func (r *rangeBuilder) buildFromPatternLike(x *ast.PatternLikeExpr) []rangePoint
 		r.err = ErrUnsupportedType.Gen("NOT LIKE is not supported.")
 		return fullRange
 	}
-	pattern := x.Pattern.GetValue().(string)
+	pattern, err := types.ToString(x.Pattern.GetValue())
+	if err != nil {
+		r.err = errors.Trace(err)
```
Comment (shenli) [inline optimizer/plan/range.go]: Will the error stop the whole query process?

File: optimizer/plan/refiner.go [hunk 2]
```diff
@@ -222,7 +223,14 @@ func (c *conditionChecker) check(condition ast.ExprNode) bool {
 		if !ast.IsPreEvaluable(x.Pattern) {
 			return false
 		}
-		patternStr := x.Pattern.GetValue().(string)
+		patternVal := x.Pattern.GetValue()
+		if patternVal == nil {
+			return false
+		}
+		patternStr, err := types.ToString(patternVal)
+		if err != nil {
```
Comment (qiuyesuifeng) [inline optimizer/plan/refiner.go]: Should we trace this error in log？

File: optimizer/plan/range.go (hunk 1 above)
Comment (coocood) [inline optimizer/plan/range.go]: Yes, the error occurs when the type of the value is unknown. In the case when we added a new type, but forgot to handle it in ToSting method.

File: optimizer/plan/refiner.go (hunk 2 above)
Comment (coocood) [inline optimizer/plan/refiner.go]: The pattern will be evaluated later, the error will be logged then.

**Comment by shenli** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**